### ISSUE TITLE
Payjoin-cli V2 returns uninstructive 400 from OHTTP relay when client ohttp_config mismatches gateway

### ISSUE THREAD

**Issue by thebrandonlucas** [payjoin/rust-payjoin#152]
Working through the [payjoin-cli](https://github.com/payjoin/rust-payjoin/tree/master/payjoin-cli#manual-end-to-end-regtest-testing) sample docs, I ran into an error when trying to generate a BIP-21 receive URI:

```
payjoin-cli receive 100
Error: HTTP request failed

Caused by:
    https://ohttp-relay.obscuravpn.io/payjoin: status code 400
```

My config.toml:
```
bitcoind_cookie="/Users/brandonlucas/Library/Application Support/Bitcoin/regtest"
bitcoind_rpchost="http://localhost:18443/wallet/alice"
pj_endpoint="https://payjo.in"
ohttp_config="AQAgQB9BwO5taMV6Skp4dl9HpFPw5SNUnnd-imKkUm8P5EIABAABAAM"
ohttp_proxy="https://ohttp-relay.obscuravpn.io/payjoin"
```

The `alice` wallet has UTXOs ready to spend from, this is all on regtest.

**Comment by DanGould** [comment]
Thanks for reporting. I am able to reproduce this error with `payjoin-cli-0.0.2-alpha`. I believe the issue is that the `ohttp_config` value does not match the output of https://payjo.in/ohttp-config. When It's set to be the same this error goes away.

The cli should make this error known so that the client can accommodate key rotation. The ohttp-relay should also be able to proxy requests for that config outside of OHTTP so the client (perhaps with HTTPS CONNECT method) can get it in an authenticated manner without revealing their IP. I'm going to change the title of this issue to reflect what I believe to be the root problem so that we can fix it.

**Comment by DanGould** [comment]
This should be fixed by #153. However, there's still an underlying question of how payjoin v2 receivers get the payjoin directory OHTTP config in an authenticated manner without revealing their IP to the directory. This could be configured by having wallet software services fetch them on a client's behalf, or better yet, [configuring the ohttp-relay to proxy requests for `application/ohttp-keys`](https://github.com/payjoin/ohttp-relay/issues/3) as a bootstrapping mechanism in addition to proxying ohttp requests (which require those ohttp-key in the first place)
